### ISSUE TITLE
Stack validation fails sometimes

### ISSUE THREAD

**Issue by darthvader88** [darthvader88/experimentOs#37]
Reproduction:

```
[parent] waiting the child to exit...
[kernel] Pid 12 will WAIT until pid 13 dies
*** DEBUG: attempt to write COW page at 0xbffff000 (addr: 0xbffffef0)
*** DEBUG: the page was not shared anymore. Making it writable.
Fork returned 0
############## I'm the child!
[thread signal]: under lock, signal_all!
[thread signal]: exit
****** [kernel thread] EXIT (pid: 8)
[validate stack] real stack page:       0xc01ff000
[validate stack] current->kernel_stack: 0xc441c000
************************ KERNEL PANIC ************************
Invalid kernel stack pointer.
File /home/travis/build/darthvader88/experimentOs/kernel/interrupts.c at line 67
Current process: 8 [KERNEL]
Interrupts: [ ]
*** KERNEL STACKTRACE ***
frame[0]: 0xc0102ee5
frame[1]: 0xc0105432
frame[2]: 0xc0103658
frame[3]: 0xc0100473
*** END STACKTRACE ***
```

**Comment by darthvader88** [comment]
Another reproduction with much more debug info:

```
 ************************ KERNEL PANIC ************************

 [validate stack] real stack page:       0xc01ff000
 [validate stack] current->kernel_stack: 0xc441c000
 Invalid kernel stack pointer.
 File /home/vlad/dev/experimentOs/kernel/interrupts.c at line 67

 Current process: 8 [KERNEL]
 Interrupts: [ ]
 Stacktrace: 0xc010303f 0xc01055b2 0xc01037cd 0xc0100473
 Registers: eflags: 0x00000402
 ss:  0x00000010, cs:  0x00000008, ds:  0x00000010, esp: 0xc01ffef4
 eip: 0xc0102f5d, eax: 0xc0102f5d, ecx: 0x00000000, edx: 0xc01ff000
 ebx: 0x00000000, ebp: 0xc01fff10, esi: 0x00000000, edi: 0x00000000
 Raw stack dump:
 0xc01ffef4: 0xc010305a 0xc4413fd4 0x00000010 0xc01fff1c
 0xc01fff04: 0xffffffff 0x00000010 0x00000000 0xc01fff50
 0xc01fff14: 0xc01055b2 0xc010abcc 0xc01ff000 0xc441c000
 0xc01fff24: 0xc010a408 0x00000043 0xc4413fd4 0xc01064f1
 0xc01fff34: 0x00000092 0x00000100 0x00000092 0x00000010
 0xc01fff44: 0x0000007b 0x00000000 0xc4413ff0 0xc01fff80
 0xc01fff54: 0xc01037cd 0xc010a408 0x00000043 0x00000002
 0xc01fff64: 0x00001000 0x00005000 0xc4424000 0xc441029c

```

**Comment by darthvader88** [comment]
This concrete type of stack validation failure has been solved by commit a4d07f5180b834dc5add173f3b13898fc9b119a8

Which makes dead kernel threads to be destroyed later (not in the scheduler) by using tasklets.
